### ISSUE TITLE
offer "--filter-position" (and perhaps --filter-symbol) via the IDE

### ISSUE THREAD

**Issue by kjx** [dafny-lang/ide-vscode#484]
The CLI has --filter-position and perhaps --filter-symbol options to drive verification.

because I have a bad habit of 10,000 line files (or rather projects in just one file - long story)
I tend to use that rather than the IDE verification.

it would be nice if I could click on a line and say "verify here" (i.e. --filter-position) 
or on (or inside) a declaration and verify that declaration (ie.. -filter-symbol)

**Comment by keyboardDrummer** [comment]
> or on (or inside) a declaration and verify that declaration (ie.. -filter-symbol)

Can't you already do that? If you set 'automatic verification' to never:

<img width="1073" alt="image" src="https://github.com/user-attachments/assets/6623f764-421b-438b-9a2e-41dfbcd61d31">

Then you get these clickable icons in the gutter:

<img width="243" alt="image" src="https://github.com/user-attachments/assets/4efe9c81-a022-4372-ba7b-12354d405ae6">


> it would be nice if I could click on a line and say "verify here" (i.e. --filter-position)

For sure. I hope to build that, but I'm currently occupied with more pressing Dafny work

**Comment by kjx** [comment]
I don't mean to hassle - and I'm fine with working with the IDE on 2/3rds or the screen and the CLI on the other.

I'd like to direct verification to a single *line* rather than a whole enclosing declaration: it's often quicker for me..

Also, no idea why the language server (does it stay around) is often slower at e.g. verifying a single method than the CLI.  Again I know it's bad practice (long story: waiting til I get to a milestone to refactor) to have a 12k line file - but that means e.g. the pre verification etc happens on the whole file each time, and the IDE appears slower to me at than the CLI...

anyway - I do appreciate what you're doing making the IDE better!

**Comment by keyboardDrummer** [comment]
> I'd like to direct verification to a single line rather than a whole enclosing declaration: it's often quicker for me..

Yes, that's what I'd like to build. However, I don't have a timeline for when I'll do that.

> Again I know it's bad practice (long story: waiting til I get to a milestone to refactor) to have a 12k line file
